core/vm/interpreter.go

```diff
@@ -304,7 +313,8 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 		err = nil // clear stop token error
 	}
 
-	return res, err
+	ret = append(ret, res...)
```

Comment:
Mabye good. But I need run some experiment with Grafana to see an impact.

Reply:
Thanks for taking a look!
For the Celeron N5100 I'm left with `github.com/ledgerwatch/erigon/crypto/bn256/cloudflare.gfpMul` dominating 47% of the CPU profiling while before I had memory.go related calls. However I must admit that the SHA/SIMD change made a larger difference than the memory pool. This machine is at 100% CPU and no iowait as it has a good nvme on a slow CPU.

Another machine syncing Ethereum mainnet on Intel 11800h using mechanical HDD with nvme lvm cache just had a minor CPU usage reduction as it is severely limited by the slow HDD IO.

If desired, I can split the changes into smaller PRs.